assets/js/components/notifications/notification.js

```diff
@@ -302,7 +302,10 @@ class Notification extends Component {
 								mdc-layout-grid__cell
 								mdc-layout-grid__cell--span-1
 							">
-								<img className="googlesitekit-publisher-win__small-image" alt="" src={ smallImage } />
+								{
+									typeof smallImage === 'string'
+										? <img className="googlesitekit-publisher-win__small-image" alt="" src={ smallImage } /> : smallImage
+								}
```

Comment:
I'm not a huge fan of this overloading of the prop; accepting very different types of arguments for props like this can be confusing, and wasn't part of the IB. I understand wanting to import the SVG directly, but the IB suggests using `${global._googlesitekitLegacyData.admin.assetsRoot}${userInputSuccessImage}` as the `src`/`smallImage` value here. It'd be great to refactor this to accept components directly in the future, but let's do that all-at-once rather than this hybrid approach that makes the API less clear.

Reply:
@kostyalmm mentioned this on Slack:

> About this PR comment, the issue is that the svg files are not hosted to the server like images are.
Exists: http://sitekit.10uplabs.com/wp-content/plugins/google-site-kit/dist/assets/images/rocket.png
Does not Exist: http://sitekit.10uplabs.com/wp-content/plugins/google-site-kit/dist/assets/svg/adsense.svg
Everywhere else SVGs are used as nodes and the import statement returns a node instead of the path, that is why I decided to overload the Notification param.

So it was my bad thinking SVGs were available as directly-referenced files. In that case, I think this is fine 🙂

Reply:
While this works, I think we should at least give it a second thought. I left a related comment here: https://github.com/google/site-kit-wp/pull/2313#discussion_r529021339

We should decide on a single approach for how we want to handle image URLs vs components like SVGs. @caseymorrisus @kostyalmm Would be great if you could find a common approach here. For both `smallImage` and `winImage` props (and in general), do we want to allow overloading this so that you can pass a component (SVG) or a string (image URL)? Or do we want to have separate props for the two alternatives? IMO the latter would be a cleaner API, but let's discuss. We should definitely implement this the same way between here and #2313.

Reply:
@felixarntz @tofumatt @caseymorrisus I suggest we use two separate props like:
```js
smallImageSrc: PropTypes.string
smallImage: PropTypes.elementType

winImageSrc: PropTypes.string
winImage: PropTypes.elementType
```
Using `PropTypes.elementType` or `WPComponent` so that we can pass additional props to the SVGs if we want to.
What do you guys think?

Reply:
@felixarntz 👍  on adding svg support.

@kostyalmm 

> Using `PropTypes.elementType` or `WPComponent` so that we can pass additional props to the SVGs if we want to.

This is a good point – the svg loader we have strips the width and height from the source svgs to allow them to be passed as props (not sure why it does that really but it does). That means that even if we wanted to accept a component type as the prop here for an svg, it would have to be wrapped for svg components to be able to specify dimensions. E.g. `winImage={ ( props ) => <SomeSVG width="300" height="200" { ...props } />  }`
So it would be a bit cleaner to pass an element, but less flexible if we wanted to support passing class names, etc to the image element from within the `Notification`.

It would be nice if we had an image loader that loaded images in the same way we load SVG so that the usage was compatible. I realize they have some differences with the way sizing works but might be nice.

Reply:
@felixarntz @aaemnnosttv @kostyalmm  Related to this conversation, I added an `ImageSVG` prop to the `Notification` component, can be found [here](https://github.com/google/site-kit-wp/blob/0ba1c62e482e89c0c40d8f263eb443a6a38247c6/assets/js/components/notifications/notification.js#L394).

As far as explicitly passing src props, I think we should stick with passing components or elements unless we're working directly within a custom image component (then a src prop is more valid).

Reply:
@caseymorrisus I agree. 
About the `ImageSVG`, I think this should be renamed to `WinImageSVG` or simply `WinImage` as using `ImageSVG` will create confusion since there is `winImage` and `smallImage`.

Reply:
@kostyalmm This is a good callout, I originally had the assumption that `smallImage` was an alternate variant of `winImage` (meaning no need to distinguish for SVGs as they scale without loss of quality) but this is incorrect.

I'll modify my PR to call this `WinImageSVG` to make this more clear which area of the notification this is affecting.

Reply:
@kostyalmm This PR still needs to be updated per the above conversation to add a `SmallImageSVG` prop and use that, instead of overloading the `smallImage` prop (similar like how in the other PR there will be a `WinImageSVG`).